### PR TITLE
PARQUET-2398: Make static variables final

### PR DESCRIPTION
**Pull request by Fokko**
Make sure you have checked _all_ steps below.

These variables should not change, therefore they should be final.

### Jira

- [ ] My PR addresses the following [Parquet Jira](https://issues.apache.org/jira/browse/PARQUET/) issues and references them in the PR title. For example, "PARQUET-1234: My Parquet PR"
  - https://issues.apache.org/jira/browse/PARQUET-XXX
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [ ] My commits all reference Jira issues in their subject lines. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain Javadoc that explain what it does

### CODE REVIEW

File: parquet-avro/src/main/java/org/apache/parquet/avro/AvroReadSupport.java [hunk 1]
```diff
@@ -44,16 +44,15 @@ public class AvroReadSupport<T> extends ReadSupport<T> {
 
   private static final Logger LOG = LoggerFactory.getLogger(AvroReadSupport.class);
 
-  public static String AVRO_REQUESTED_PROJECTION = "parquet.avro.projection";
+  public static final String AVRO_REQUESTED_PROJECTION = "parquet.avro.projection";
```
Comment (amousavigourabi) [inline parquet-avro/src/main/java/org/apache/parquet/avro/AvroReadSupport.java:47]: All of these newly `final` fields should be excluded by the japicmp configuration.

File: pom.xml [hunk 2]
```diff
@@ -596,6 +596,8 @@
 
               <!-- Due to the removal of deprecated methods -->
               <exclude>org.apache.parquet.arrow.schema.SchemaMapping</exclude>
+
+              <exclude>org.apache.parquet.hadoop.ParquetFileReader.PARQUET_READ_PARALLELISM</exclude>
```
Comment (amousavigourabi) [inline pom.xml:600]:
```suggestion
              <!-- Make static variables final -->
              <exclude>org.apache.parquet.avro.AvroReadSupport#AVRO_REQUESTED_PROJECTION</exclude>
              <exclude>org.apache.parquet.avro.AvroReadSupport#AVRO_DATA_SUPPLIER</exclude>
              <exclude>org.apache.parquet.hadoop.ParquetFileReader#PARQUET_READ_PARALLELISM</exclude>
```

**Review by amousavigourabi — CHANGES_REQUESTED** [review]

**Comment by wgtmac** [comment]
@Fokko Do you still want to check in this for 1.14.0?

**Comment by Fokko** [comment]
@wgtmac Yes please! Let me fix the PR

File: pom.xml (hunk 2 above)
Comment (Fokko) [reply pom.xml]: Thanks :)

**Review by wgtmac — APPROVED** [review]

File: pom.xml [hunk 3]
```diff
@@ -599,6 +599,11 @@
               <!-- Due to the removal of deprecated methods -->
               <exclude>org.apache.parquet.arrow.schema.SchemaMapping</exclude>
 
+              <!-- Make static variables final -->
+              <exclude>org.apache.parquet.avro.AvroReadSupport#AVRO_REQUESTED_PROJECTION</exclude>
```
Comment (wgtmac) [inline pom.xml:603]: These are indeed breaking changes from the perspective of API but I think it makes sense to add final keyword to them.

**Pull request merged**